inst/tutorials/chapter4/standard_errors_standalone.Rmd

```diff
@@ -0,0 +1,111 @@
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(plotly)
+```
+
+## Sampling App
+
+This app simulates the drawing of a random sample of 20 observation from a population, with the underlying data generating process (DGP):
+
+$$Y = 5 + 2X + \nu, \quad \text{with:}$$
+$$X \sim \mathcal{N}(5, 2^2)\quad \text{and} \quad \nu \sim \mathcal{N}(0, 5^2)$$ 
+
+(so $\nu$ is the sampling variability, or noise)
+
+
+```{r, echo=FALSE}
+shinyUI(fluidPage(
+  actionButton("new", "New Draw!"),
+  plotlyOutput("sample"),
+  fluidRow(column(width = 6, plotlyOutput("hist_a")),
+           column(width = 6, plotlyOutput("hist_b")))
+))
+```
+
+
+```{r, context="server"}
+
+# Generate 500 x/y draws
+
+set.seed(19)
+df <- as.data.frame(matrix(rep(0, 20*1000), c(20, 1000)))
+
+for (i in seq(1, 999, 2)){
+  df[i] <- rnorm(20, 5, 2)
+  df[i+1] <- 5 + 2*df[i] + rnorm(20, 0, 5)
+}
+
+ab <- as.data.frame(matrix(rep(NA, 2*999), c(999, 2)))
+colnames(ab) <- c("a", "b")
+
+for (i in seq(1, 999, 2)){
+  ab[i, "a"] <- cov(df[[i]], df[[i+1]])/var(df[[i]])
+}
+for (i in seq(1, 999, 2)){
+  ab[i, "b"] <- mean(df[[i+1]]) - ab$a[i] * mean(df[[i]])
+}
+
+
+counter <- reactiveValues(j = 3) #define the counter j as a reactiveValue (stored in the list "counter")
+
+observeEvent(input$new, { #dynamically update counter j, handle j > 999
+  if (counter$j > 999){
+    counter$j <- 3
+  } else {
+    counter$j <- (counter$j + 2)
+  }
+})
+
+
+output$sample <- renderPlotly({
+  
+  slope_current <- (cov(df[[counter$j]], df[[counter$j + 1]]))/var(df[[counter$j]])
+  inter_current <- mean(df[[counter$j + 1]]) - slope_current*mean(df[[counter$j]])
+  
+  slope_past <- (cov(df[[counter$j - 2]], df[[counter$j - 1]]))/var(df[[counter$j - 2]])
+  inter_past <- mean(df[[counter$j - 1]]) - slope_past*mean(df[[counter$j - 2]])
+  
+  #Plot
+  
+  plot_ly() %>%
+    add_markers(x = df[[counter$j]], y = df[[counter$j+1]],
+                opacity = 1, marker = list(color = 'blue'),
+                name = "Current Draw") %>%
+    add_markers(x = df[[counter$j-2]], y = df[[counter$j-1]],
+                opacity = .5, marker = list(color = 'blue'),
+                name = "Past Draw") %>%
+    
+    add_lines(x = c(0, 10), y = c(inter_current, inter_current + 10*slope_current),
+              name = "Current Best Fit", line = list(color = 'green')) %>%
+    add_lines(x = c(0, 10), y = c(inter_past, inter_past + 10*slope_past),
+              name = "Past Best Fit", opacity = .5, line = list(color = 'green')) %>%
+    add_lines(x = c(0, 10), y = c(5,  25),
+              name = "Population Regression Line", opacity = .25,
+              line = list(color = 'grey', dash = 'dash')) %>%
+    
+    layout(title = "Fitted Sample of n = 20 Draws From the Population",
+           xaxis = list(range = c(-1, 10), showgrid = F, title = "X"),
+           yaxis = list(range = c(0, 30), showgrid = F, title = "Y"))
+})
+
+output$hist_a <- renderPlotly({
```

Comment:
when i do this it seems that it's the wrong way around, i.e.the intercepts are centered around 2 and the slope at 5 which doesn't correspond at least with the title.

Reply:
ah yes, I fixed it in the standalone version of the app (PR #36)